tests/test_relationships.py

```
@@ -0,0 +1,221 @@
+        with self.client.application.app_context():
+            product = Product.query.get(product_id)
+            assert product is not None
+            return sorted([subcategory.id for subcategory in product.subcategories])
+
+    def _category_product_ids_via_subcategories(self, category_id):
+        with self.client.application.app_context():
+            category = Category.query.get(category_id)
+            assert category is not None
+            return sorted({product.id for subcategory in category.subcategories for product in subcategory.products.all()})
+
+    def _assert_related_collection(self, resp, key, expected_ids=[], status_code=200):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)